dotnet/samples/KernelSyntaxExamples/Example87_GeminiVision.cs

```diff
@@ -0,0 +1,106 @@
+{
+    [Fact]
+    public async Task RunAsync()
+    {
+        this.WriteLine("======== Gemini chat with vision ========");
+
+        await GoogleAIGeminiAsync();
+        await VertexAIGeminiAsync();
+    }
+
+    private async Task GoogleAIGeminiAsync()
+    {
+        this.WriteLine("===== Google AI Gemini API =====");
+
+        string geminiApiKey = TestConfiguration.GoogleAI.ApiKey;
+        string geminiModelId = "gemini-pro-vision";
+
+        if (geminiApiKey is null)
+        {
+            this.WriteLine("Gemini credentials not found. Skipping example.");
+            return;
+        }
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddGoogleAIGeminiChatCompletion(
+                modelId: geminiModelId,
+                apiKey: geminiApiKey)
+            .Build();
+
+        var chatHistory = new ChatHistory();
+        var chatCompletionService = kernel.GetRequiredService<IChatCompletionService>();
+
+        chatHistory.AddUserMessage(new ChatMessageContentItemCollection
+        {
+            new TextContent("What’s in this image?"),
+            // Google AI Gemini API requires the image to be in base64 format, doesn't support URI
+            // You have to always provide the mimeType for the image
+            new ImageContent(new BinaryData(EmbeddedResource.ReadStream("sample_image.jpg")),
+                metadata: new Dictionary<string, object?> { { "mimeType", "image/jpeg" } }),
+        });
+
+        var reply = await chatCompletionService.GetChatMessageContentAsync(chatHistory);
+
+        WriteLine(reply.Content);
+    }
+
+    private async Task VertexAIGeminiAsync()
+    {
+        this.WriteLine("===== Vertex AI Gemini API =====");
+
+        string geminiApiKey = TestConfiguration.VertexAI.ApiKey;
+        string geminiModelId = "gemini-pro-vision";
+        string geminiLocation = TestConfiguration.VertexAI.Location;
+        string geminiProject = TestConfiguration.VertexAI.ProjectId;
+
+        if (geminiApiKey is null || geminiLocation is null || geminiProject is null)
+        {
+            this.WriteLine("Gemini vertex ai credentials not found. Skipping example.");
+            return;
+        }
+
+        Kernel kernel = Kernel.CreateBuilder()
+            .AddVertexAIGeminiChatCompletion(
+                modelId: geminiModelId,
+                apiKey: geminiApiKey,
+                location: geminiLocation,
+                projectId: geminiProject)
+            .Build();
+
+        var chatHistory = new ChatHistory();
+        var chatCompletionService = kernel.GetRequiredService<IChatCompletionService>();
+
+        chatHistory.AddUserMessage(new ChatMessageContentItemCollection
+        {
+            new TextContent("What’s in this image?"),
+            // Vertex AI Gemini API supports both base64 and URI format
+            // You have to always provide the mimeType for the image
+            new ImageContent(new BinaryData(EmbeddedResource.ReadStream("sample_image.jpg")),
```

Comment:
If the mimeType is always required for an image, should we model it through a dedicated 'string MimeType {get; set;}' property on the ImageContent type and allow to set it via a parameter of class constructors?

Reply:
@SergeyMenshykh I think this topic is more complicated. I raised the topic of ImageContent changes at https://github.com/microsoft/semantic-kernel/issues/4759 but did not get a concrete answer.

Reply:
BinaryData has the `MediaType`property, do we really need to have an extra metadata for this?

```diff
-new ImageContent(new BinaryData(EmbeddedResource.ReadStream("sample_image.jpg")),
+new ImageContent(new BinaryData(EmbeddedResource.ReadStream("sample_image.jpg"), "image/jpeg")),
```

Reply:
my mistake, fixed